applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -391,8 +393,11 @@ static void scan_start_trigger_fn(struct k_work *w)
 		scan_counter = 0;
 		scan_start();
 	} else {
-		k_delayed_work_submit(&scan_start_trigger,
-				      K_MSEC(SCAN_TRIG_CHECK_MS));
+		int err = k_work_reschedule(&scan_start_trigger,
```

Comment:
FWIW I believe this can return negative only if `SCAN_TRIG_CHECK_MS` is zero; is that a legitimate configuration?

See also https://github.com/zephyrproject-rtos/zephyr/pull/35337 inspired by this idiom.

Reply:
you are actually right; check here has little sense as `SCAN_TRIG_CHECK_MS` cannot be zero (we validate that)

Reply:
removed the unnecessary checks